personal_finances/user/user_persistence.py

```diff
@@ -0,0 +1,107 @@
+from functools import cache
+import logging
+import boto3
+import os
+from typing import Any, List
+import boto3.dynamodb
+import boto3.dynamodb.table
+from pydantic import BaseModel, ValidationError
+from schwifty.exceptions import SchwiftyException
+from personal_finances.user.user_auth_exceptions import (
+    UserNotFound,
+    InvalidIban,
+    InvalidDynamoResponse,
+)
+from schwifty import IBAN
+
+
+LOGGER = logging.getLogger(__name__)
+USER_ID_MIN_LEN = 6
+USER_PASSWORD_MIN_LEN = 6
+
+
+class User(BaseModel):
+    userId: str
+    password: str
+    ibanList: List[str]
+
+
+@cache
+def _get_user_table() -> Any:
+    return boto3.resource("dynamodb").Table(
+        os.environ["INFINEAT_DYNAMODB_USER_TABLE_NAME"]
+    )
+
+
+def get_user_password(userId: str) -> str:
+
+    response = _get_user_table().get_item(
+        Key={"userId": userId},
+    )
+    if "Item" not in response:
+        raise UserNotFound(f"User not found: {userId}")
+
+    try:
+        user = User(**response["Item"])
+    except ValidationError as e:
+        LOGGER.error(f"Data validation error: {e}")
+        raise InvalidDynamoResponse("Invalid data format from DynamoDB")
+
+    return user.password
+
+
+def get_user_ibans(userId: str) -> List[str]:
+    table = _get_user_table()
+
+    try:
+        LOGGER.info("Getting user IBAN list...")
+        response = table.get_item(Key={"userId": userId})
+
+        if "Item" not in response:
+            raise UserNotFound(f"User not found: {userId}")
+
+        user = User(**response["Item"])
+        iban_list = user.ibanList
+
+        LOGGER.info(f"User IBAN list:{iban_list}")
+        return iban_list
+
+    except ValidationError as e:
+        LOGGER.error(f"Data validation error: {e}")
+        raise InvalidDynamoResponse("Invalid data format from DynamoDB")
```

Comment:
add `from e` at the end of the exception to keep the context from previous exceptions

Reply:
new issue created for that: Keep the context from previous exceptions in the user persistence #53